cocos/animation/core/pose-allocator.ts

```diff
@@ -0,0 +1,148 @@
+import { assertIsTrue } from '../../core/data/utils/asserts';
+import { Pose } from './pose';
+import { TransformArray } from './transform-array';
+
+const MAX_POSE_PER_PAGE = 8;
+
+export class PoseAllocator {
```

Comment:
The name `PosePageAllocator` may be more appropriate, and it should be noted that the allocator type is a page allocator

Reply:
This class in my opnion should not expose its implementation-no matter it use pool or anything else, it should be treated as a black-box from external's view. So I advise to keep the name.

Reply:
It's OK if we won't implement another allocator